### PR TITLE
Improve error message when user passes nonexistent plugin

### LINKED ISSUE #6000: Raise an informative exception when a non-existent plugin is requested
## 🐛 Bug

If you invoke napari from the command line with the `--plugin` flag with a plugin that does not exist, napari responds with a generic error message that doesn't disclose that the plugin doesn't exist. This can lead someone into thinking a plugin is malfunctioning, when it simply hasn't been installed correctly (e.g., because the `pip` command got hijacked by pyenv 😡 )

## To Reproduce

```bash
(napari) ➜  ~ napari --plugin not_a_plugin /path/to/data
Traceback (most recent call last):
  File "/Users/bennettd/mambaforge/envs/napari/bin/napari", line 10, in <module>
    sys.exit(main())
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/__main__.py", line 446, in main
    _run()
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/__main__.py", line 311, in _run
    viewer._window._qt_viewer._qt_open(
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/_qt/qt_viewer.py", line 754, in _qt_open
    self.viewer.open(
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/components/viewer_model.py", line 931, in open
    self._add_layers_with_plugins(
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/components/viewer_model.py", line 1134, in _add_layers_with_plugins
    layer_data, hookimpl = read_data_with_plugins(
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/plugins/io.py", line 68, in read_data_with_plugins
    res = _npe2.read(paths, plugin, stack=stack)
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/plugins/_npe2.py", line 54, in read
    raise e from e
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/napari/plugins/_npe2.py", line 50, in read
    layer_data, reader = read_get_reader(npe1_path, plugin_name=plugin)
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/npe2/io_utils.py", line 66, in read_get_reader
    return _read(
  File "/Users/bennettd/mambaforge/envs/napari/lib/python3.9/site-packages/npe2/io_utils.py", line 169, in _read
    raise ValueError(
ValueError: Plugin 'not_a_plugin' was selected to open ['/path/to/data'], but returned no data.
```

## Expected behavior

If I try to use a plugin that doesn't exist, napari should tell me that the plugin doesn't exist.

## Environment

```
napari: 0.4.16
Platform: macOS-12.5-arm64-arm-64bit
System: MacOS 12.5
Python: 3.9.16 | packaged by conda-forge | (main, Feb 1 2023, 21:38:11) [Clang 14.0.6 ]
Qt: 5.15.6
PyQt5: 5.15.7
NumPy: 1.24.3
SciPy: 1.9.1
Dask: 2023.6.0
VisPy: 0.10.0

OpenGL:
- GL version: 2.1 Metal - 76.3
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1470x956, scale 2.0

Plugins:
- console: 0.0.8
- napari-ome-zarr: 0.5.2
- napari-svg: 0.1.7
- scikit-image: 0.4.16
```

### PR DESCRIPTION
**Pull request by DragaDoncila**
# Fixes/Closes

Closes #6000

# Description
This PR updates the error message shown to users when they pass a nonexistent reader plugin to `viewer.open`. It adds npe2 readers to the list of available readers, and adds a specific message for when no readers are available.

# How has this been tested?
I was sad to find there's no tests for `read_data_with_plugins` that don't just mock it :< I can write a test tomorrow morning but don't have the fortitude for it right now.

### CODE REVIEW

File: napari/plugins/io.py [hunk 1]
```diff
@@ -100,13 +100,20 @@ def read_data_with_plugins(
...
         if plugin not in plugin_manager.plugins:
-            names = {i.plugin_name for i in hook_caller.get_hookimpls()}
+            names = set(_npe2.get_readers().keys()).union(
+                {i.plugin_name for i in hook_caller.get_hookimpls()}
+            )
+            err_helper = (
+                "No readers are available. Do you have any plugins installed?"
+                if not names
+                else f"\nNames of plugins offering readers are: {names}."
+            )
```
Comment (Czaki) [inline napari/plugins/io.py]: We always have `napari_builtins` present. So maybe we should do `if len(names) < 1`?

Comment (Carreau) [reply napari/plugins/io.py]: Should those strings be in `trans._` calls ?

**Comment by DragaDoncila** [comment]
Put this in draft mode as I'm writing tests for the affected function. Should have tests done tomorrow 🙏

File: napari/plugins/io.py (hunk 1 above)
Comment (jni) [reply napari/plugins/io.py]: Addressed both of these.

**Review by jni — APPROVED** [review]

**Pull request merged**